lib/ui/pages/note_camera/note_camera.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:camera/camera.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:get/get.dart';
+import 'package:flutter_app/src/store/picture_storage.dart';
+
+class NoteCamera extends StatefulWidget {
+  final String courseId;
+
+  const NoteCamera({Key? key, required this.courseId}) : super(key: key);
+
+  @override
+  State<NoteCamera> createState() => _NoteCameraState();
+}
+
+class _NoteCameraState extends State<NoteCamera> with WidgetsBindingObserver {
+  CameraController? controller;
+  bool _isCameraInitialized = false;
+  late List<CameraDescription> cameras;
+
+  double _zoom = 1.0;
+
+  _NoteCameraState();
+
+  void onNewCameraSelected(CameraDescription cameraDescription) async {
+    final previousCameraController = controller;
+
+    final CameraController cameraController = CameraController(
+      cameraDescription,
+      ResolutionPreset.high,
+      imageFormatGroup: ImageFormatGroup.jpeg,
+    );
+
+    await previousCameraController?.dispose();
+
+    if (mounted) {
+      setState(() {
+        controller = cameraController;
+      });
+    }
+    cameraController.addListener(() {
+      if (mounted) setState(() {});
+    });
+
+    try {
+      await cameraController.initialize();
+    } on CameraException catch (e) {
+      throw Exception('Error initializing camera: $e');
+    }
+
+    if (mounted) {
+      setState(() {
+        _isCameraInitialized = controller!.value.isInitialized;
+      });
+    }
+  }
+
+  void takePicture() async {
+    final image = await controller?.takePicture();
+    String? path = image?.path;
+    if (path != null) PictureStorage.takePictureToStorage(widget.courseId, path);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    SystemChrome.setEnabledSystemUIMode(SystemUiMode.manual, overlays: []);
+
+    cameras = Get.find<List<CameraDescription>>();
+    onNewCameraSelected(cameras[0]);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    final CameraController? cameraController = controller;
```

Comment:
Remove the type when you declare with `final`.

```suggestion
    final cameraController = controller;
```